topic-configuration/README.adoc

```diff
@@ -0,0 +1,689 @@
+|===
+--
+
+=== Cleanup
+
+These properties control the cleanup processing of the log.
+
+Log segment size::
++
+--
+Log segment size is the size of the log segment files. Log processing actions, such as deletion and compaction, operate on log segments, so a larger setting results in fewer files but less frequent log processing.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Integer
+
+h|Default value
+|`1 GB`
+
+h|Kafka property name
+|`segment.bytes`
+|===
+--
+
+Segment time::
++
+--
+Segment time is the amount of time after which the current log segment is rolled even if the segment file is not full. This property ensures that the topic cleanup policy can delete or compact old messages as needed.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Long
+
+h|Default value
+|`604800000 ms` (7 days)
+
+h|Kafka property name
+|`segment.ms`
+|===
+--
+
+Segment jitter time::
++
+--
+Segment jitter time is the maximum delay for log segment rolling. This delay prevents bursts of log segment rolling activity.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Long
+
+h|Default value
+|`0 seconds`
+
+h|Kafka property name
+|`segment.jitter.ms`
+|===
+--
+
+File delete delay::
++
+--
+File delete delay is the amount of time that a file is retained in the system before the file is deleted.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Long
+
+h|Default value
+|`60000 ms` (1 minute)
+
+h|Kafka property name
+|`file.delete.delay.ms`
+|===
+--
+
+Preallocate log segment files::
++
+--
+Preallocate log segment files determines whether to preallocate the file on disk when creating a new log segment. This property ensures sufficient disk space for log segments.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Boolean
+
+h|Default value
+|`Disabled`
+
+h|Kafka property name
+|`preallocate`
+|===
+--
+
+=== Index
+
+These properties control the indexing of the log.
+
+Index interval size::
++
+--
+Index interval size is the number of bytes between each index entry to its offset index. The default setting indexes a message about every 4096 bytes. More indexing enables reads to be closer to the exact position in the log but makes the index larger.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Integer
+
+h|Default value
+|`4 KB`
+
+h|Kafka property name
+|`index.interval.bytes`
+|===
+--
+
+Segment index size::
++
+--
+Segment index size is the size of the index that maps offset to file positions.
+
+.Property attributes
+[cols="25%,75%"]
+|===
+h|Editable
+|No
+
+h|Type
+|Integer
+
+h|Default value
+|`10 MB`
+
+h|Kafka property name
+|`segment.index.bytes`
+|===
+--
+
+=== Flush
+
+These properties control the frequency of the flushing of the log.
+
+Flush interval messages::
++
+--
+Flush interval messages is the number of messages between each data flush to the log.
```

Comment:
I'm not clear on this. Are we flushing from the log cache? So it's how many messages are received before forcing a flush of a topic’s messages to disk?

Reply:
Why's the default in time?

Reply:
My understanding is that there's a data flush mechanism to the log itself within the topic, and this property dictates how often by count of messages that flush occurs.

And yes, great eye! It's a bug. Davide mentioned that above and I filed an issue for it. Details in this comment above: https://github.com/redhat-developer/app-services-guides/pull/144#discussion_r639548941